File: infrastructure/src/main/java/com/study/platform/domain/notification/infrastructure/FailedNotificationRepositoryAdapter.java
package com.study.platform.domain.notification.infrastructure;

import com.study.platform.domain.notification.model.FailedNotification;
import com.study.platform.domain.notification.model.FailedNotificationRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class FailedNotificationRepositoryAdapter implements FailedNotificationRepository {

    private final FailedNotificationJpaRepository failedNotificationJpaRepository;

    @Override
    public FailedNotification save(FailedNotification failedNotification) {
        return failedNotificationJpaRepository.save(failedNotification);
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/domain/notification/infrastructure/FailedNotificationRepositoryAdapter.kt
package com.study.platform.domain.notification.infrastructure

import com.study.platform.domain.notification.model.FailedNotification
import com.study.platform.domain.notification.model.FailedNotificationRepository
import org.springframework.stereotype.Repository

@Repository
class FailedNotificationRepositoryAdapter(
    private val failedNotificationJpaRepository: FailedNotificationJpaRepository
) : FailedNotificationRepository {

    override fun save(failedNotification: FailedNotification): FailedNotification =
        failedNotificationJpaRepository.save(failedNotification)
}
